runtime-modules/content/src/benchmarks/benchmarking.rs

```diff
@@ -2249,147 +2261,1096 @@ benchmarks! {
+    // ============================ NFT - BUY NOW =====================================
+    // ================================================================================
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
+    // - curator owned channel
+    // - curator number is max
+    // - curator has max number of agent permissions
+    // - channel has max size:
+    //   - all feature paused (except necessary ones for extr to succeed)
+    //   - max channel assets
+    //   - max collaborators
+    // - video has max size
+    //   - max video assets
+    // - nft limits are set
+    // - nft owner is channel owner
+    // INPUT COMPLEXITY
+    sell_nft {
+        let (
+            video_id,
+            (curator_account_id, actor, channel_id, _)
+        ) = setup_worst_case_scenario_mutable_video::<T>(
+            Some(T::MaxNumberOfAssetsPerVideo::get()),
+            T::StorageBucketsPerBagValueConstraint::get().max() as u32,
+        )?;
+
+        let price = BUY_NOW_PRICE.into();
+
+        let _ = setup_idle_nft::<T>(
+            curator_account_id.clone(),
+            actor,
+            video_id,
+            false,
+        )?;
+
+        set_all_channel_paused_features::<T>(channel_id);
+        let origin = RawOrigin::Signed(curator_account_id.clone());
+
+    }: _ (origin, video_id, actor, price)
+        verify {
+            assert!(matches!(Pallet::<T>::video_by_id(video_id).nft_status, Some(Nft::<T> {
+                transactional_status: TransactionalStatus::<T>::BuyNow(price),
+                ..
+            })));
+        }
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
+    // - curator owned channel
+    // - curator number is max
+    // - curator has max number of permissions
+    // - channel has all features paused except necessary ones
+    // - channel has max assets
+    // - NFT owner == channel owner
+    // INPUT COMPLEXITY
+    cancel_buy_now {
+        let (
+            video_id,
+            (curator_account_id, actor, channel_id, _)
+        ) = setup_worst_case_scenario_mutable_video::<T>(
+            Some(T::MaxNumberOfAssetsPerVideo::get()),
+            T::StorageBucketsPerBagValueConstraint::get().max() as u32,
+        )?;
+
+        let (_, to_member) = member_funded_account::<T>(MEMBER_IDS[1]);
+        let price = BUY_NOW_PRICE.into();
+
+        let _ = setup_nft_in_buy_now::<T>(
+            curator_account_id.clone(),
+            actor,
+            video_id,
+            false,
+            price,
+        )?;
+
+        set_all_channel_paused_features::<T>(channel_id);
+        let origin = RawOrigin::Signed(curator_account_id.clone());
+    }: _ (origin, actor, video_id)
+        verify {
+            assert!(matches!(Pallet::<T>::video_by_id(video_id).nft_status, Some(Nft::<T> {
+                transactional_status: TransactionalStatus::<T>::Idle,
+                ..
+            })));
+        }
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
```

Comment:
State complexity comment not updated and still missing conditions like:
- video should have max number of assets
- curator group should have max number of permissions per level
- channel should have max number of collaborators